lib/splitclient-rb/engine/api/events.rb

```diff
@@ -22,10 +22,12 @@ def post
             'SplitSDKVersion' => event_slice[0][:m][:s]
           )
 
-          if (200..299).include? result.status
-            SplitIoClient.configuration.logger.debug("Events reported: #{event_slice.size}") if SplitIoClient.configuration.debug_enabled
+          if response.success?
+            SplitLogger.log_if_debug("Events reported: #{event_slice.size}")
           else
-            SplitIoClient.configuration.logger.error("Unexpected status code while posting events: #{result.status}")
+            SplitLogger.log_error("Unexpected status code while posting events: #{response.status}." \
+            " - Check your API key and base URI")
+            raise 'Split SDK failed to connect to backend to post events'
```

Comment:
Is there any reason for throwing here instead of just logging? Ideally we won't to avoid raising exceptions that may under any circumstance propagate to the user and bring down the application. My understanding is that this is within the synchronizer portion of the sdk. Is this guarded by a try/catch (or the ruby equivalent)? Unless there's a very specific reason to throw i'd rather just log the error as we are doing on every other sdk. cc: @NicoZelaya @mmelograno

Reply:
agreed with you

Reply:
They're catching that from a parent function.